### PR TITLE
explorer view event action icon improvements

### LINKED ISSUE #2139: UX: Create a "Copy entire event" button
Would be nice to have a button int he event details that copies the entire event as a jsonl blob
`[{"key1:"value1","key2":"value2"...}]`

### LINKED ISSUE #2270: Get link to a single event
**Is your feature request related to a problem? Please describe.**
It would be nice to be able to get a link to a single event, which can be embedded into external documentation. 

**Describe the solution you'd like**
A "Get Link" button on the event row, which copies a link to the event to the clipboard. When you access the link, you end up on the Explore page that shows only the single event. 

The link could likely use the Opensearch internal docid, as an implementation detail?

**Describe alternatives you've considered**
You can apparently do this also with existing functionality, creating a search that returns the particular event, saving that search, and using the link to the saved search. This is a bit cumbersome.

**Additional context**

### LINKED ISSUE #2371: Have a link to event button in frontend-ng
https://github.com/google/timesketch/pull/2305

Make that happen in frontend-ng

### PR DESCRIPTION
**Pull request by jkppr**
This PR will change the placement of the event action icons with the explore view in the new frontend-ng and add related new functionalities.  
- The new placement of the icons on the beginning of the event entry is part of the improved UI/UX for the event explorer view.
- Additionally this PR adds three new functionalities that were requested in multiple issues:
  - A `Copy link to even` function that copies the direct link to this particular event into the clipboard.
  - A `Copy event data as JSON` function that copies the full event details as JSON string into the clipboard.
  - An `infoSnackBar(message)` function that will trigger a blue snackBar with shorter timeout as a quick feedback info to the user.

image

**Closing issues**
- closes #2270 
- closes #2371 
- closes #2139 
- closes #2462

### CODE REVIEW

File: timesketch/frontend-ng/src/components/Explore/EventActionMenu.vue [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      ApiClient.getEvent(this.sketch.id, this.event._index, this.event._id)
+        .then((response) => {
+          let fullEvent = response.data.objects
+          let eventJSON = JSON.stringify(fullEvent, null, 3)
+          navigator.clipboard.writeText(eventJSON)
+          this.infoSnackBar('Event JSON copied to clipboard')
+        })
+        .catch((e) => {
+          this.errorSnackBar('Failed to load JSON into the clipboard')
+          console.error(e)
```
Comment (jaegeral) [inline timesketch/frontend-ng/src/components/Explore/EventActionMenu.vue:67]: this should not be needed as we catch it centrally in the API class

File: timesketch/frontend-ng/src/components/Explore/EventActionMenu.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        })
+    },
+    copyEventUrlToClipboard() {
+      try {
+        let eventUrl = window.location.origin + this.$route.path + '?q=_id:' + this.event._id
+        navigator.clipboard.writeText(eventUrl)
+        this.infoSnackBar('Event URL copied to clipboard')
+      } catch (error) {
+        this.errorSnackBar('Failed to load Event URL into the clipboard')
+        console.error(error)
```
Comment (jaegeral) [inline timesketch/frontend-ng/src/components/Explore/EventActionMenu.vue:77]: same as above

**Review by jaegeral — COMMENTED** [review]
some smaller nits, overall looks pretty solid to me

File: timesketch/frontend-ng/src/components/Explore/EventActionMenu.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
+<!--
+Copyright 2022 Google Inc. All rights reserved.
```
Comment (berggren) [inline timesketch/frontend-ng/src/components/Explore/EventActionMenu.vue:2]:
```suggestion
Copyright 2023 Google Inc. All rights reserved.
```

**Review by berggren — APPROVED** [review]
Apart from a couple of previos comments, LGTM

File: timesketch/frontend-ng/src/components/Explore/EventActionMenu.vue (hunk 1 above)
Comment (jkppr) [reply timesketch/frontend-ng/src/components/Explore/EventActionMenu.vue:67]: To my understanding we only catch API errors centrally. So worst case the user gets two snackbars telling them that something didn't work. As discussed offline, I'll keep it this way for now to also catch other errors that can happen and inform the user for a better UX.

**Pull request merged**